assets/src/editor/events/ticket-assignments-manager/hooks/use-counts-manager.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * External imports
+ */
+import { useMemo, useCallback } from '@wordpress/element';
+import { reduce, cloneDeep, isEmpty } from 'lodash';
+import { __ } from '@eventespresso/i18n';
+
+// @todo, when the wp.data.useSelect hook is available, we can utilize that in
+// this custom hook and then we'll be able to remove the withSelect hoc usage
+// in the composed `TicketAssignmentsManagerModal` component.
+
+const useCountsManager = (
+	dateEntities,
+	ticketEntities,
+	assignedCounts,
+	assignedState,
+) => {
+	const getMessage = useCallback( ( whichAreEmpty ) => {
+		let message;
+		const onlyTicketsAreEmpty = whichAreEmpty.dates === 0 && whichAreEmpty.tickets > 0;
+		const onlyDatesAreEmpty = whichAreEmpty.dates > 0 && whichAreEmpty.tickets === 0;
+		const anyAreEmpty = whichAreEmpty.dates > 0 || whichAreEmpty.tickets > 0;
+
+		switch ( true ) {
+			case ( dateEntities.length === 1 && onlyDatesAreEmpty ) ||
+			( ticketEntities.length === 1 && onlyDatesAreEmpty ) :
+				message = __(
+					'Event Dates must always have at least one Ticket assigned to them. If the current assignment is not correct, assign the correct Ticket first, then remove others as required.',
+					'event_espresso'
+				);
+				break;
+			case ( ticketEntities.length === 1 && anyAreEmpty ) ||
+			( dateEntities.length === 1 && onlyTicketsAreEmpty ) :
+				message = __(
+					'Tickets must always have at least one Event Date assigned to them. If the current assignment is not correct, assign the correct Event Date first, then remove others as required.',
+					'event_espresso'
+				);
+				break;
+			default:
+				message = __(
+					'Event Dates must always have at least one Ticket assigned to them but one or more of the Event Dates below does not have any. Please correct the assignments for the highlighted cells.',
+					'event_espresso'
+				);
+		}
+		return message;
+	}, [ dateEntities, ticketEntities ] );
+
+	const isEmptyAssignedState = useCallback(
+		() => isEmpty( assignedState.assigned ) && isEmpty( assignedState.removed ),
+		[ assignedState ]
+	);
+
+	const [ hasNoAssignments, noAssignmentsMessage, updatedAssignmentsCounts ] = useMemo( () => {
+		let howManyEmpty = 0;
+		const whichAreEmpty = { dates: 0, tickets: 0 };
+		const updatedCounts = cloneDeep( assignedCounts );
+		// first let's see if any of the current counts are 0 (but only not
+		// already toggled as empty)!
+		// instead of using whichAreEmpty as a ref, use assignedState (whether its empty or not) to decide whether we
+		// do this here or not. If assigned state is empty then we can do it.
+		if ( isEmptyAssignedState() ) {
```

Comment:
Previously I was using `whichAreEmpty` as a ref for deciding whether to do this initial short-circuit calc.  That was very fragile because relying on the ref to keep counts through state changes was super fragile.  All that was really needed here is to run this short circuit path if `assignedState` is empty (meaning the original assignedCounts could be used instead)